packages/edge/ios/src/RCTAEPExperienceEventDataBridge.m

```diff
@@ -0,0 +1,58 @@
+/*
+ Copyright 2021 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+#import "RCTAEPExperienceEventDataBridge.h"
+
+@implementation RCTAEPExperienceEventDataBridge
+
+static NSString* const XDM_DATA_KEY = @"xdmData";
+static NSString* const DATA_KEY = @"data";
+static NSString* const DATASET_IDENTIFIER_KEY = @"datasetIdentifier";
+
+static NSString* const TYPE_KEY = @"type";
+static NSString* const PAYLOAD_KEY = @"payload";
+
++ (AEPExperienceEvent *)experienceEventFromDictionary: (nonnull NSDictionary *) dict {
+    
+    if (!dict || [dict isEqual:[NSNull null]]) {
+            return nil;
+        }
+    
+    NSDictionary *xdmdata = [dict objectForKey:XDM_DATA_KEY];
+    NSDictionary *data = [dict objectForKey:DATA_KEY];
+    NSString *datasetIdentifier = [dict objectForKey:DATASET_IDENTIFIER_KEY];
+    
+    if (!xdmdata) {
+        return nil;
+    }
+    
+    xdmdata = [[dict objectForKey:XDM_DATA_KEY] isKindOfClass:[NSDictionary class]] || ![dict objectForKey:XDM_DATA_KEY] ? xdmdata  : nil;
+    data =  [[dict objectForKey:DATA_KEY] isKindOfClass:[NSDictionary class]] || ![dict objectForKey:DATA_KEY] ? data : nil;
+    datasetIdentifier = [[dict objectForKey:DATASET_IDENTIFIER_KEY] isKindOfClass:[NSString class]] ? datasetIdentifier : nil;
```

Comment:
L29->39 Did you intend to check the type before assignment? You can rewrite this to something like:
```
NSDictionary *xdmdata = [[dict objectForKey:XDM_DATA_KEY] isKindOfClass:[NSDictionary class]] ? [dict objectForKey:XDM_DATA_KEY] : nil;

if (!xdmdata) {
    return nil;
}
NSDictionary *data = [[dict objectForKey:DATA_KEY] isKindOfClass:[NSDictionary class]] ? [dict objectForKey:DATA_KEY] : nil;
NSString *datasetIdentifier = [[dict objectForKey:DATASET_IDENTIFIER_KEY] isKindOfClass:[NSString class]] ? [dict objectForKey:DATASET_IDENTIFIER_KEY] : nil;
```